### ISSUE TITLE
Tile loading issues

### ISSUE THREAD

**Issue by VKatilis** [bolrog/d2dx#176]
Noticed that (mostly in act II and Act V) you would have blacked out tiles near the screen edge (or as discovered recently, elements of the interface see image), no matter what the resolution is. Happens when moving towards the respective edge. Much worse in motion as those triangles flicker across the edges. Not sure whether it is motion prediction or D2 multi red causing it

![6BF30C6B-71D3-4749-A8FD-33A4BC13AD5C](https://user-images.githubusercontent.com/112778889/188285133-5176f032-5800-49ca-a3e9-865c68fe43a4.jpeg)

**Comment by VKatilis** [comment]
UPDATE:

This is definitely caused by the motion prediction. Turning it off fixes the issue. Which really sucks cause otherwise it works really well

**Comment by gamedevsam** [comment]
This actually seems to be related to perspective mode for me. Turning off perspective mode fixed the issue.

**Comment by VKatilis** [comment]
Definitely not perspective mode. If anything, turning it off made it even more noticeable

**Comment by eleriaqueen** [comment]
Hey, for me the artifacts around the edges of the screen and open inventory/stats panel are acceptable. I'm talking about the "garbage" that appears at the edges of the screen when moving rapidly.

(These artifacts most likely would be hard to fix as you'd need D2DX to render more cells than needed around the screen in order for the prediction to have data to work with thus preventing them)

Now about the black tiles issue...

I'm not able to get tiles to look as jarring as in the picture from VKatillis though.

The most visible artifacts I'm able to reproduce are these :

![Undrawn cells](https://user-images.githubusercontent.com/4411178/193410707-5eee6b88-1241-48e5-a96a-f3bfed6b5d4a.png)

Using freshly compiled D2DX from the original repo with these settings : 
```
[window]
scale=2			# range 1-3
position=[-1,-1]	# if [-1,-1] the window will be centered, otherwise placed at the explicit position given here
frameless=false         # if true, the window frame (caption bar etc) will be removed

[game]
size=[-1,-1]		# if [-1,-1] d2dx will decide a suitable game size, otherwise will use the size given here
filtering=0             # if 0, will use high quality filtering (sharp, more pixelated)
                        #    1, will use bilinear filtering (blurry)
                        #    2, will use catmull-rom filtering (higher quality than bilinear)

#
# Opt-outs from default D2DX behavior
#
[optouts]
noclipcursor=true	 # if true, will not lock the mouse cursor to the game window
nofpsfix=false		 # if true, will not apply the basic fps fix (precludes high fps support)
noresmod=false		 # if true, will not apply the built-in D2HD resolution mod (precludes widescreen support)
nowide=false		 # if true, will not choose a widescreen resolution (if noresmod is true, this does nothing)
nologo=false		 # if true, will not display the D2DX logo on the title screen
novsync=false		 # if true, will not use vertical sync
noaa=false		 # if true, will not apply anti-aliasing to jagged edges
nocompatmodefix=true	 # if true, will not block the use of "Windows XP compatibility mode"
notitlechange=false	 # if true, will not change the window title text
nomotionprediction=false # if true, will not run the game graphics at high fps
```

**Comment by VKatilis** [comment]
The “garbage”  you’re talking about is the same issue.

**Comment by eleriaqueen** [comment]
@VKatilis can you test this version with updated external SGD2FreeRes ? I see less culling issues with it.
https://github.com/eleriaqueen/d2dx/releases/tag/unofficial-v02

**Comment by Vahsify** [comment]
@eleriaqueen It works fine with 1.09d, but MapHack for 1.09d don't recognise version of Diablo II with that glide3x.dll. With 0.99.529 MapHack works fine.

**Comment by eleriaqueen** [comment]
@Vahsify weird... are you sure it's MapHack's error message you're seing ?
I would need more detail to maybe know what's going on.

**Comment by Vahsify** [comment]
@eleriaqueen Yes. Error "Unknown game version" if I click "Install/update" and nothing more. But with Bolrog's 0.99.529 (and other versions) MapHack works fine.

**Comment by eleriaqueen** [comment]
@Vahsify I enabled issues tracker and added one with the problem you've described

**Comment by VKatilis** [comment]
> @VKatilis can you test this version with updated external SGD2FreeRes ? I see less culling issues with it. https://github.com/eleriaqueen/d2dx/releases/tag/unofficial-v02

Just launched with this - it does seem less noticeable, but it is still there unfortunately

**Comment by Vahsify** [comment]
> > @VKatilis can you test this version with updated external SGD2FreeRes ? I see less culling issues with it. https://github.com/eleriaqueen/d2dx/releases/tag/unofficial-v02
> 
> Just launched with this - it does seem less noticeable, but it is still there unfortunately

"Extended the tile culling even more, to lessen the tile culling problem in D2DX with motion prediction enabled. Is not guaranteed to fully fix the tile culling problem."

https://github.com/mir-diablo-ii-tools/SlashGaming-Diablo-II-Free-Resolution/releases

**Comment by CaptainRay1993** [comment]
You can try [my build](https://github.com/CaptainRay1993/d2dx/releases/tag/0.99.529c) which is similar to @eleriaqueen build but with the latest SGD2FreeRes update.

**Comment by wasdwdsa** [comment]
With lastest SGD2FreeRes.dll (3.0.3.1), black triangles don't appear at top or bottom edges, still flicker (less noticeable than before) at left or right edge when character/inventory page is shown.
![2](https://user-images.githubusercontent.com/47906639/200118289-824c16bb-2139-4dba-9654-c5f0a7248097.jpg)
